internal/command/jsonformat/plan.go

```diff
@@ -453,6 +461,13 @@ func resourceChangeComment(resource jsonplan.ResourceChange, action plans.Action
 		case jsonplan.ResourceInstanceDeleteBecauseEachKey:
 			buf.WriteString(fmt.Sprintf("\n  # (because key [%s] is not in for_each map)", resource.Index))
 		}
+		if len(resource.Deposed) != 0 {
+			// Some extra context about this unusual situation.
```

Comment:
I'd like to know more about how this situation is unusual. Maybe an example would help?

Reply:
I altered the comment to contain a better explanation.
But I'll add a more extended version here:
Let's say you have a resource set with a `create_before_destroy` [lifecycle configuration](https://opentofu.org/docs/language/meta-arguments/lifecycle/#syntax-and-arguments). You do a change that requires replacing your resource. Because the `create_before_destroy` is set, tofu first sets the existing resource as deposed (so it can destroy it later), creates the new instance of the resource, and then fails when it tries to destroy the deposed instance.
In this scenario, the deposed instance will be saved as a leftover in the state. On the next `apply`, tofu will try to destroy/forget it, and it'll mention it specifically in the plan.